apps/condo/domains/user/schema/ConfirmPhoneAction.js

```diff
@@ -0,0 +1,245 @@
+const { ConfirmPhoneAction: ConfirmPhoneActionGQL, generateSmsCode } = require('@condo/domains/user/utils/serverSchema')
+
+const { 
+    CONFIRM_PHONE_ACTION_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_VERIFICATION_FAILED, 
+    CONFIRM_PHONE_SMS_CODE_MAX_RETRIES_REACHED, 
+} = require('@condo/domains/user/constants/errors')
+
+const { COUNTRIES, RUSSIA_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { SMS_VERIFY_CODE_MESSAGE_TYPE } = require('@condo/domains/notification/constants')
+
+const { 
+    SMS_CODE_LENGTH, 
+    SMS_CODE_TTL, 
+    CONFIRM_PHONE_ACTION_EXPIRY, 
+    CONFIRM_PHONE_SMS_MAX_RETRIES,
+} = require('@condo/domains/user/constants/common')
+
+const ConfirmPhoneAction = new GQLListSchema('ConfirmPhoneAction', {
+    schemaDoc: 'User confirm phone actions is used before registration starts',
+    fields: {
+        dv: DV_FIELD,
+        sender: SENDER_FIELD,
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            kmigratorOptions: { null: true, unique: false },
+            hooks: {
+                resolveInput: ({ resolvedData }) => {
+                    return normalizePhone(resolvedData['phone'])
+                },
+            },
+        },
+        token: {
+            schemaDoc: 'Unique token to complete confirmation',
+            type: Text,
+            isUnique: true,
+            isRequired: true,
+        },
+        smsCode: {
+            schemaDoc: 'Last sms code sent to user',
+            type: Integer,
+            length: SMS_CODE_LENGTH,
+        },
+        smsCodeRequestedAt: {
+            schemaDoc: 'Time when sms code was requested',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        smsCodeExpiresAt: {
+            schemaDoc: 'Time when sms code becomes not valid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        retries: {
+            schemaDoc: 'Number of times sms code input from user failed',
+            type: Integer,
+            defaultValue: 0,
+        },
+        isPhoneVerified: {
+            schemaDoc: 'Phone verification flag. User verify phone by access to secret sms message',
+            type: Checkbox,
+            defaultValue: false,
+        },
+        requestedAt: {
+            schemaDoc: 'DateTime when confirm phone action was started',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        expiresAt: {
+            schemaDoc: 'When confirm phone action becomes invalid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        completedAt: {
+            schemaDoc: 'When confirm phone action was completed',
+            type: DateTimeUtc,
+            isRequired: false,
+        },
+
+    },
+    plugins: [uuided(), softDeleted(), historical()],
```

Comment:
add versioned(), tracked()

Reply:
tracked is not usable as anonymous user only take part in confirm phone

Reply:
Some admin users also have access to this model. And it's better to track their changes